### PR TITLE
refactor: abstract db away

### LINKED ISSUE #33: Abstract MongoDB/ElasticSearch/NATS interactions into common packages
Abstract DB logic (interfaces) into common packages so that we can write unit tests.

### PR DESCRIPTION
**Pull request by maxlivinci**
resolve #33 
partof #11 

## Dependency Inversion Principle

High-level modules should not depend on low-level modules; both should depend on abstractions.

image

### CODE REVIEW

File: services/validation/internal/service/validation_service_test.go [hunk 1]
```diff
@@ -25,4 +24,18 @@ func TestGetLinkedSchemas(t *testing.T) {
...
+	_, ok = getLinkedSchemas(map[string]interface{}{"linked_schemas": []interface{}{"https://ic3.dev/test2.json"}})
+	assert.Equal(t, true, ok)
+
+	linkedSchemas, ok := getLinkedSchemas(map[string]interface{}{"linked_schemas": []interface{}{"https://ic3.dev/test2.json"}})
+	assert.Equal(t, "https://ic3.dev/test2.json", linkedSchemas[0])
+}
+
+func TestGetSchemaURL(t *testing.T) {
+	url := getSchemaURL("test1.json")
+	assert.Equal(t, "/schemas/test1.json.json", url)
```
Comment (geoffturk) [inline services/validation/internal/service/validation_service_test.go:40]: `.json.json`?

Comment (maxlivinci) [reply services/validation/internal/service/validation_service_test.go]: Sorry. Should not include `.json` in the input.

Comment (maxlivinci) [reply services/validation/internal/service/validation_service_test.go]: We don't have the test environment variables and I have created an issue for it. #53

**Review by geoffturk — APPROVED** [review]
LGTM 🚀

**Pull request merged**